### ISSUE TITLE
tf.keras.losses.categorical_crossentropy and binary_crossentropy (and other losses) only works for channels_last layout networks

### ISSUE THREAD

**Issue by dd1923** [tensorflow/tensorflow#39230]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Pip3
- TensorFlow version (use command below): v2.1
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**

Looks like in a bizarre api design the constructor of neither `tf.keras.losses.categorical_crossentropy` nor `tf.keras.losses.binary_crossentropy` take an axis parameter as input. An axis specifies along which dimensions one wants to calculate these losses. Current implementation simply assumes last axis. Without a way to specify a different axis in the constructor, these can only be used for networks where `image_data_layout` is set to `channels_last`. `channels_first` network are out of luck because they require different axis. 

Also inconsistent api design is evident in `tf.keras.backend.categorical_crossentropy`. This backend version of categorical_crossentropy does that an axis as an input in its constructor. Though one can use this as a loss function as a workaround, it still does not solve the binary cross entropy problem. And also results in an overall inconsistent api.

The `tf.keras.losses.categorical_crossentropy` is actually calling the backend version but since axis was an optional parameter, it gets set by default to -1 all the time. See [this line in the losses.py code](https://github.com/tensorflow/tensorflow/blob/e5bf8de410005de06a7ff5393fafdf832ef1d4ad/tensorflow/python/keras/losses.py#L971)

**Describe the expected behavior**

 `tf.keras.losses.categorical_crossentropy` and `tf.keras.losses.binary_crossentropy` (and may be all other loss functions) should take an axis parameter as input. So that these losses can be used in channels_first setups.

**Standalone code to reproduce the issue**
None needed. See the documentation of these loss functions and see the source code link above.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
@dd1923 

Will it be possible to share the colab link or simple standalone code to reproduce the issue reported here.It helps us in localizing the issue faster.Thanks!

**Comment by dd1923** [comment]
@ravikyram Did you read the post above? What do you want to reproduce? The documentation and the code link I mentioned clearly shows it. This is a case of bad api design.

**Comment by dd1923** [comment]
@pavithrasv

**Comment by dd1923** [comment]
@jhseu @pavithrasv when do you think this would be assigned to someone?
